package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,355 @@
+An example of this is shown below, using the MSD computed in the example above. The segment between :math:`\tau = 20` and :math:`\tau = 80` is used to demonstrate selection of an MSD segment.
+
+    >>> from scipy.stats import linregress as lr
+    >>> start_time = 20
+    >>> start_index = start_time/timestep
+    >>> end_time = 80
+    >>> end_index = end_time/timestep
+    >>> linear_model = lr(lagtimes[start_index:end_index], msd[start_index:end_index])
+    >>> slope = linear_model.slope
+    >>> error = linear_model.rvalue
+    >>> D = slope * 1/(2*MSD.dim_fac) #dim_fac is 3 as we computed a 3D msd ('xyz')
+
+We have now computed a self diffusivity!
+
+
+Notes
+_____
+
+There are several factors that must be taken into account when setting up and processing trajectories for computation of self diffusivities.
+These include specific instructions around simulation settings, using unwrapped trajectories and maintaining relativley small elapsed time between saved frames. 
+Additionally corrections for finite size effects are sometimes employed along with varied means of estimating errors.
+The reader is directed to the following review, which describes many of the common pitfalls [Maginn2019]_. There are other ways to compute self diffusivity including from a Green-Kubo integral. At this point in time these methods are beyond the scope of this module
+
+
+References
+----------
+
+.. [Maginn2019] Maginn, E. J.; Messerly, R. A.; Carlson, D. J.; Roe, D. R.; Elliott, J. R. Best Practices for Computing Transport Properties 1. Self-Diffusivity and Viscosity from Equilibrium Molecular Dynamics [Article v1.0]. Living J. Comput. Mol. Sci. 2019, 1 (1).
+.. [Calandri2011] Calandrini, V.; Pellegrini, E.; Calligari, P.; Hinsen, K.; Kneller, G. R. NMoldyn-Interfacing Spectroscopic Experiments, Molecular Dynamics Simulations and Models for Time Correlation Functions. Collect. SFN 2011, 12, 201–232.
+.. [SO2015] https://stackoverflow.com/questions/34222272/computing-mean-square-displacement-using-python-and-fft
+
+Classes and Functions
+---------------------
+
+.. autoclass:: MeanSquaredDisplacement
+    :members:
+
+"""
+
+from __future__ import division, absolute_import
+from six.moves import zip
+
+import os
+import errno
+import warnings
+import bz2
+import functools
+
+import numpy as np
+from numpy.fft import fft,ifft
+import logging
+import MDAnalysis
+
+class MeanSquaredDisplacement(object):
+    r"""Class representing Mean Squared Displacement
+
+    Parameters
+    ----------
+    u : Universe
+        An MDAnalysis :class:`Universe`
+    selection : str
+        An MDAnalysis selection string
+    msd_type : str
+        The dimensions to use for the msd calculation:
+        one of ('xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z')
+    fft : bool
+        Use a fast FFT based algorithm for computation of the MSD
+
+    Returns
```

Comment:
no returns required

Reply:
I assume this was for the class docstring rather than the method docstrings